src/main/java/TestBase.java

```diff
@@ -0,0 +1,31 @@
+import dataprovider.DataProviderXML;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeSuite;
+
+import java.util.concurrent.TimeUnit;
+
+public class TestBase extends DataProviderXML {
```

Comment:
It will be better to create Helper class which contains method readXML(File sourceFile), and use it in dataProvider.
This data provider used only in one test class, so you can store provider method directly here. 
Please avoid redundant nesting.  (Prefer composing instead nesting, for detailed information read about Strategy Pattern (Design Patterns, Freeman, Freeman))